vignettes/data-combined-plotting.Rmd

```diff
@@ -0,0 +1,182 @@
+simulated data (represented by lines).
+
+```{r}
+plotIndividualTimeProfile(myDataCombined)
+```
+
+## Observed versus simulated scatter plot 
+
+Observed versus simulated plots show allow to assess how far simulated results 
+are from observed values.
+
+```{r}
+plotObservedVsSimulated(myDataCombined)
+```
+
+The identity line represents perfect correspondance of simulated values with the 
+observed ones. By default, a "two-fold" are is marked by the dashed lines. 
+The "x-fold" range is defined as values that are `x`-fold above and `1/x`-fold 
+below the observed onces. The user can specify multiple ranges by the `foldDistance` 
+argument.
+
+```{r}
+plotObservedVsSimulated(myDataCombined, foldDistance = c(1.3, 2))
+```
+
+## Residuals versus time or vs simulated scatter plot
+
+Residual plots show if there is a systematic bias in how the simulation represents values either in high-concentration or low-concentration regions, or, alternatively, in early or late time periods.
+
+```{r}
+plotResidualsVsSimulated(myDataCombined)
+```
+
+
+```{r}
+plotResidualsVsTime(myDataCombined)
+```
+
+## Customizing plot aesthetics
+
+The aesthetics for plots can be customized using the `DefaultPlotConfiguration` class, which provides various class members that can be used to modify the *look* of the plot.
+
+```{r}
+myPlotConfiguration <- DefaultPlotConfiguration$new()
+
+# Define x units
+myPlotConfiguration$xUnit <- ospUnits$Time$s
+# Define y units
+myPlotConfiguration$yUnit <- ospUnits$`Concentration [mass]`$`µg/l`
+# Change y axis scaling to logarithmic
+myPlotConfiguration$yAxisScale <- tlf::Scaling$log
+
+myPlotConfiguration$title <- "Example: Customizing a Plot"
+myPlotConfiguration$subtitle <- "Using `DefaultPlotConfiguration` class"
+myPlotConfiguration$caption <- "Source: `Aciclovir` data"
+
+myPlotConfiguration$legendPosition <- tlf::LegendPositions$outsideRight
+```
+
+This configuration class can be passed to all plotting functions:
+
+```{r}
+plotIndividualTimeProfile(myDataCombined, myPlotConfiguration)
+```
+
+## Creating multi-panel plots
+
+Each of the `plotXXX()` returnsan a `ggplot2` object. Lets create different plots 
+from the same `DataCombined` and store them as variables.
+
+```{r}
+indivProfile <- plotIndividualTimeProfile(myDataCombined, myPlotConfiguration)
+obsVsSim <- plotObservedVsSimulated(myDataCombined, myPlotConfiguration)
+resVsSim <- plotResidualsVsSimulated(myDataCombined)
+resVsTime <- plotResidualsVsTime(myDataCombined)
+```
+
+These plots can be combined into a multi-panel figure using the `PlotGridConfiguration` 
+and then used with `plotGrid()` function to create a figure.
+
+```{r}
+plotGridConfiguration <- PlotGridConfiguration$new()
+plotGridConfiguration$tagLevels <- "a"
+plotGridConfiguration$title <- "Multiple plots in one figure"
+
+plotGridConfiguration$addPlots(plots = list(indivProfile, obsVsSim, resVsSim, resVsTime))
+
+plotGrid(plotGridConfiguration)
+```
```

Comment:
This example creates a 2x2 plot matrix, but I don't see where those values are set (probably the defaults).
It would be good to show the X/Y number of plots in the plot matrix **explicitely**

Reply:
Also I guess plotGrid can have just one (global) figure caption?
If yes: would be good to include it into the example as well

Reply:
> Also I guess plotGrid can have just one (global) figure caption?

`plotGridConfiguration$title <- "Multiple plots in one figure"`

is it, now?

Reply:
Added an example with only one row.